ethcore/src/spec/spec.rs

```diff
@@ -511,11 +487,18 @@ impl Spec {
 	/// Alter the value of the genesis state.
 	pub fn set_genesis_state(&mut self, s: PodState) -> Result<(), Error> {
 		self.genesis_state = s;
-		let _ = self.run_constructors(
+		let (root, _) = run_constructors(
+			&self.genesis_state,
+			&self.constructors,
+			&*self.engine,
+			self.author,
+			self.timestamp,
+			self.difficulty,
 			&Default::default(),
 			BasicBackend(journaldb::new_memory_db()),
 		)?;
 
+		self.state_root_memo = root;
```

Comment:
`run_constructors` no longer mutates `self.state_root_memo`, so we need to set the root manutally